python/Ganga/testlib/GangaUnitTest.py

```diff
@@ -62,49 +65,32 @@ def start_ganga(gangadir_for_test, extra_opts=[]):
     Ganga.Runtime._prog = Ganga.Runtime.GangaProgram(argv=this_argv)
     Ganga.Runtime._prog.parseOptions()
 
-    # Determine if ganga has actually finished initializing...
-    # This is here to protect against the startGanga being called on an
-    # initialized ganga environment
-    try:
-        do_config = not Ganga.Utility.Config.Config._after_bootstrap
-    except:
-        do_config = True
-
     # For all the default and extra options, we set the session value
     from Ganga.Utility.Config import setConfigOption
     for opt in default_opts + extra_opts:
         setConfigOption(*opt)
 
-    if do_config:
-        # Perform the configuration and bootstrap steps in ganga
-        logger.info("Parsing Configuration Options")
-        Ganga.Runtime._prog.configure()
-        logger.info("Initializing")
-        Ganga.Runtime._prog.initEnvironment()
-    else:
-        from Ganga.Runtime.Repository_runtime import startUpRegistries
-        from Ganga.Utility.Config import getConfig
-        if getConfig('Configuration')['AutoStartReg']:
-            startUpRegistries()
-
-        # The queues are shut down by the atexit handlers so we need to start them here
-        from Ganga.Core.GangaThread.WorkerThreads import startUpQueues
-        startUpQueues()
-
-        # We need to test if the internal services need to be reinitialized
-        from Ganga.Core.InternalServices import Coordinator
-        if not Coordinator.servicesEnabled:
-            # Start internal services
-            logger.info("InternalServices restarting")
-
-            from Ganga.Core.InternalServices.Coordinator import enableInternalServices
-            enableInternalServices()
-        else:
-            logger.info("InternalServices still running")
+    # Perform the configuration and bootstrap steps in ganga
+    logger.info("Parsing Configuration Options")
+    Ganga.Runtime._prog.configure()
```

Comment:
It certainly seems to be. The only problem with it seemed to be needing to reset the config handlers which is now being done in the test teardown.